src/main/java/de/tum/in/www1/artemis/web/rest/ProgrammingExerciseParticipationResource.java

```diff
@@ -272,6 +274,38 @@ public List<CommitInfoDTO> getCommitHistoryForParticipationRepo(@PathVariable lo
         return programmingExerciseParticipationService.getCommitInfos(participation);
     }
 
+    /**
+     * GET /programming-exercise/{exerciseID}/commit-history/{repositoryType} : Get the commit history of a programming exercise repository. The repository type can be TEMPLATE or
+     * SOLUTION or TESTS.
+     * Here we check is at least a teaching assistant for the exercise.
+     *
+     * @param exerciseID     the id of the exercise for which to retrieve the commit history
+     * @param repositoryType the type of the repository for which to retrieve the commit history
+     * @return A list of commitInfo DTOs with the commits information of the repository
+     */
+    @GetMapping("programming-exercise/{exerciseID}/commit-history/{repositoryType}")
+    @EnforceAtLeastTutor
+    public List<CommitInfoDTO> getCommitHistoryForTemplateSolutionOrTestRepo(@PathVariable long exerciseID, @PathVariable RepositoryType repositoryType) {
+        boolean isTestRepository = repositoryType.equals(RepositoryType.TESTS);
+        ProgrammingExerciseParticipation participation;
+        if (!repositoryType.equals(RepositoryType.SOLUTION) && !repositoryType.equals(RepositoryType.TEMPLATE) && !repositoryType.equals(RepositoryType.TESTS)) {
+            throw new BadRequestAlertException("Invalid repository type", ENTITY_NAME, "invalidRepositoryType");
+        }
+        else if (repositoryType.equals(RepositoryType.TEMPLATE)) {
+            participation = programmingExerciseParticipationService.findTemplateParticipationByProgrammingExerciseId(exerciseID);
+        }
+        else {
+            participation = programmingExerciseParticipationService.findSolutionParticipationByProgrammingExerciseId(exerciseID);
+        }
+        participationAuthCheckService.checkCanAccessParticipationElseThrow(participation);
+        if (isTestRepository) {
+            return programmingExerciseParticipationService.getCommitInfosTestRepo(participation);
+        }
+        else {
+            return programmingExerciseParticipationService.getCommitInfos(participation);
+        }
```

Comment:
you define isTestRepository here but later use repositoryType.equals(RepositoryType.TESTS) again.
I think it would be better to define these boolean values for all (template, solution, tests) and then consistently use them in this method